src/components/SpiceList.js

```diff
@@ -2,13 +2,43 @@ import React from 'react'
 import SpiceItem from './SpiceItem'
 
 class SpiceList extends React.Component {
+
   state = {
     fourStarOnly: false,
     search: ""
   }
 
-  renderSpices() {
-    return this.props.spices.map(spice => (
+  handleChange = () => {
+    this.setState((prevState) => ({
+      fourStarOnly: !prevState.fourStarOnly
+    }))
+  }
+
+  onChange = (evt) => {
+    let searchTerm = evt.target.value.toLowerCase()
+    this.setState((prevState) => ({
+      search: searchTerm
+    }))
+  }
+
+  filteredSpices = () => {
+    if (this.state.fourStarOnly) {
+      return this.props.spices.filter(spice => spice.rating >= 4)
+    } else {
+      return this.props.spices
+    }
+  }
+
+  filterBySearch = () => {
+    let searchSpice = this.state.search
+    if (searchSpice.length === 0) {
+      return this.filteredSpices()
+    } else {
+    return this.filteredSpices().filter(spice => spice.notes.toLowerCase().includes(searchSpice))}
+  }
+
+  renderSpices = () => {
+    return this.filterBySearch().map(spice => (
       <SpiceItem key={spice.id} spice={spice} />
     ))
   }
```

Comment:
Nice work getting the multiple filters set up! If you want to see a slightly different approach, here's an approach of chaining multiple filters together:

```js
filteredSpices = () => {
  return this.props.spices
      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true)
      .filter(spice => spice.notes.toLowerCase().includes(this.state.search))
}
```

Reply:
oh sweet! Never occurred to chain them both back to back like that! Thanks!!!